include/envoy/config/subscription.h

```diff
@@ -79,6 +80,10 @@ class Subscription {
    * be passed to std::set_difference, which must be given sorted collections.
    */
   virtual void updateResources(const std::set<std::string>& update_to_these_names) PURE;
+
+  virtual void updateResourcesViaAliases(const std::set<std::string>&) {
```

Comment:
Can you add Doxygen docs for this?

Reply:
If I am perceiving the intention correctly, I think you might also want to change the name. `updateResources()` is kind of misnamed; it's not about getting updated resources, but rather updating the set of resource names we care about. Similarly, it looks like this one is about adding to a list of aliases, not updating resources.

...so I guess I'm really saying now would be a good time to give `updateResources()` a better name. Can I suggest something like either `updateResourceInterest` or `updateSubscriptionInterest`?

As for the aliases one, since it looks like it's more of a blanket addition than a "replace with this set" (which is how `updateResources` currently works), I would suggest like `addAliases` or `addResourceAliases`.